src/store/addressBookSlice.ts

```diff
@@ -33,6 +33,20 @@ export const addressBookSlice = createSlice({
       state[chainId][address] = name
     },
 
+    upsertMultichainAddressBookEntry: (
+      state,
+      action: PayloadAction<{ chainIds: string[]; address: string; name: string }>,
+    ) => {
+      const { chainIds, address, name } = action.payload
+      if (name.trim() === '') {
+        return
+      }
+      chainIds.forEach((chainId) => {
+        if (!state[chainId]) state[chainId] = {}
+        state[chainId][address] = name
+      })
+    },
+
```

Comment:
This is mostly identical with `upsertAddressBookEntry`. I tried replacing it with the following but for some reason the unit test for the slice fails.

```
const { chainIds, address, name } = action.payload

chainIds.forEach((chainId) => {
  upsertAddressBookEntry({ chainId, address, name })
})
```

Reply:
Although it might be an anti-pattern to dispatch another action from a reducer.